lib/ClangImporter/ImportDecl.cpp

```diff
@@ -1426,11 +1426,10 @@ namespace {
               // Create a typealias for this CF typedef.
               TypeAliasDecl *typealias = nullptr;
               typealias = Impl.createDeclWithClangNode<TypeAliasDecl>(
```

Comment:
Would it make sense to make `createDeclWithClangNode` always set the `AccessLevel` to `importer::convertClangAccess(Decl->getAccess())` and simplify all the call sites? Or are there exceptions where we want to do something different?

Reply:
This is a good point. My patch does leave quite a few exceptions, but I'll need to look through them again to figure out whether they are justified.

At a high level, these are some of the current exceptions:
- Objective-C decls, which should have `AS_none`, so most cases should be fine. However, the one exception is overridable Objective-C members, which need `AccessLevel::Open` instead of `AccessLevel::Public`.
- `GenericTypeParamDecl`, which seems to need `AccessLevel::Public`.
- `ParamDecl`, which seems to need `AccessLevel::Private`.

I'm tempted to leave this kind of clean up to a later follow-up patch, because otherwise I'll need to touch those cases too and enlarge this patch.

Reply:
I am fine with doing this in a follow-up.